app/utils/Utils.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import java.net.{URI, URLEncoder}
+
+import scala.util.Try
+
+object Utils {
+  def arrivalId(location: String): Try[String] =
+    Try(new URI(location).getPath.split("/").last)
```

Comment:
:thumbsup:

I think LocationHeader needs to be changed so that the exception propagates up the call hierarchy. 

object LocationHeader {
  def parse(location: String): LocationHeader = {
    Try(new URI(location).getPath.split("/").last)
  }
}

And then Success and Failure is handled in the controller.

Reply:
That might work better, go for it.